src/java.base/share/classes/java/net/SocketOptions.java

```diff
@@ -47,224 +47,167 @@
 
-    /** Same as above. This option is introduced so that the behaviour
-     *  with IP_MULTICAST_IF will be kept the same as before, while
-     *  this new option can support setting outgoing interfaces with either
-     *  IPv4 and IPv6 addresses.
+    /**
+     * Set the outgoing interface on which to send multicast packets. Useful on hosts with
+     * multiple network interfaces, where applications want to use other than the system default.
+     * This option supports setting outgoing interfaces with either IPv4 and IPv6 addresses.
      *
-     *  NOTE: make sure there is no conflict with this
      * @see MulticastSocket#setNetworkInterface(NetworkInterface)
      * @see MulticastSocket#getNetworkInterface()
      * @since 1.4
      */
     @Native public static final int IP_MULTICAST_IF2 = 0x1f;
 
     /**
-     * This option enables or disables local loopback of multicast datagrams.
-     * This option is enabled by default for Multicast Sockets.
+     * See {@link StandardSocketOptions#IP_MULTICAST_LOOP} for description of this socket option.
      * @since 1.4
      */
-
     @Native public static final int IP_MULTICAST_LOOP = 0x12;
 
     /**
-     * This option sets the type-of-service or traffic class field
-     * in the IP header for a TCP or UDP socket.
+     * See {@link StandardSocketOptions#IP_TOS} for description of this socket option.
      * @since 1.4
      */
-
     @Native public static final int IP_TOS = 0x3;
 
     /**
-     * Specify a linger-on-close timeout.  This option disables/enables
-     * immediate return from a <B>close()</B> of a TCP Socket.  Enabling
-     * this option with a non-zero Integer <I>timeout</I> means that a
-     * <B>close()</B> will block pending the transmission and acknowledgement
-     * of all data written to the peer, at which point the socket is closed
-     * <I>gracefully</I>.  Upon reaching the linger timeout, the socket is
-     * closed <I>forcefully</I>, with a TCP RST. Enabling the option with a
-     * timeout of zero does a forceful close immediately. If the specified
-     * timeout value exceeds 65,535 it will be reduced to 65,535.
-     * <P>
-     * Valid only for TCP: SocketImpl
+     * See {@link StandardSocketOptions#SO_LINGER} for description of this socket option.
      *
      * @see Socket#setSoLinger
      * @see Socket#getSoLinger
      */
     @Native public static final int SO_LINGER = 0x0080;
 
-    /** Set a timeout on blocking Socket operations:
-     * <PRE>
-     * ServerSocket.accept();
-     * SocketInputStream.read();
-     * DatagramSocket.receive();
-     * </PRE>
-     *
-     * <P> The option must be set prior to entering a blocking
-     * operation to take effect.  If the timeout expires and the
-     * operation would continue to block,
-     * <B>java.io.InterruptedIOException</B> is raised.  The Socket is
-     * not closed in this case.
+    /**
+     * Set a timeout on blocking {@code Socket} operations:
```

Comment:
This option is also used to set and get the timeout value.

Reply:
Done